packages/common/core-interfaces/src/error.ts

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryProperties } from "./index";
+
+/**
+ * Error types the Fluid Framework may report.
+ */
+export const FluidErrorTypes = {
+	/**
+	 * Some error, most likely an exception caught by runtime and propagated to container as critical error
+	 */
+	genericError: "genericError",
+
+	/**
+	 * Throttling error from server. Server is busy and is asking not to reconnect for some time
+	 */
+	throttlingError: "throttlingError",
+
+	/**
+	 * Data loss error detected by Container / DeltaManager. Likely points to storage issue.
+	 */
+	dataCorruptionError: "dataCorruptionError",
+
+	/**
+	 * Error encountered when processing an operation. May correlate with data corruption.
+	 */
+	dataProcessingError: "dataProcessingError",
+
+	/**
+	 * Error indicating an API is being used improperly resulting in an invalid operation.
+	 */
+	usageError: "usageError",
+} as const;
+export type FluidErrorTypes = typeof FluidErrorTypes[keyof typeof FluidErrorTypes];
+
+/**
+ * Base interface for all errors and warnings emitted by the Fluid Framework.
```

Comment:
Nit: the change from "at container level" to "by the Fluid Framework" seems like it might not be accurate...? Errors thrown by loader/drivers won't necessarily inherit from this (they couldn't in the past). Might be where we want to get at some point but I think it'd be safer to keep the previous docs.

Reply:
The intention is to start unifying the common aspects of errors in this package. Right now this is only used by the container/loader level, but that should change.

Reply:
And that makes sense, just leaves me slightly concerned that a host-app dev will run into this and assume they can `try {} catch(err) { }` around loader calls assuming everything caught in there will be an `IErrorBase` starting now, when it might only become true several releases from now.

Reply:
I think it's okay to keep this change here as is. Currently all consumers are using the original (albeit deprecated) one from container-definitions still.

Reply:
> I think it's okay to keep this change here as is. Currently all consumers are using the original (albeit deprecated) one from container-definitions still.

The only issue is that this is what we are re-exporting out of `container-definitions`, so existing users will see the updated docs. We should probably add a `@remarks` block noting the current status vs the long term intention.